src/csm_testing/lib/iuf_classes.py

```diff
@@ -0,0 +1,277 @@
+#
+# MIT License
+#
+# (C) Copyright 2024 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+"""
+Defining class for API calls
+"""
+import os
+import base64
+from urllib.error import HTTPError
+import requests
+from kubernetes import client, config  # pylint: disable=import-error
+from keycloak import KeycloakOpenID  # pylint: disable=import-error
+
+
+class AuthException(Exception):
+    """A wrapper for raising an AuthException exception."""
+
+
+class Auth:  # pylint: disable=missing-class-docstring
```

Comment:
We do not want to resolve the pylint errors just by adding these disable statements. The disable statements are intended to be used for cases like:
1. pylint is complaining about things that are not really errors. The best example I can think of for this is the test which imports the `rados` package. That package is not installed when we run pylint, so we get an import error, but we know that this package will be available on the actual systems. This is because the RPM itself requires that this is installed in the system Python environment.
2. pylint is complaining about something that is being done deliberately and we have a good reason not to change it.
3. There are a small number of cases where the pylint complaint is valid, but we know we will never address it. The only example of this I can think of is that we generally add `disable=missing-function-docstring` for the `main()` methods in the `__main__.py` files. This is because that function is the main body of the test or tool, and the module-level docstring should already contain a description of the test or tool logic. So repeating it in the `main()` function docstring would mostly be redundant. I suppose this could also be the case for extremely small, simple, self-explanatory functions.